src/python/Services/ReqMgr/ReqMgrReader.py

```diff
@@ -6,53 +6,383 @@
+            self.logger.error(str(error))
+            if try_cache:
+                self.logger.info("Will try to get workload cache")
+                return self._getWorkflowSchema(
+                    wf, drop_null=drop_null, make_copy=make_copy, from_cache=True
+                )
+
+    def _getWorkflowSchema(
+        self,
+        wf: str,
+        drop_null: bool = False,
+        make_copy: bool = False,
+        from_cache: bool = False,
+    ) -> dict:
+        """
+        The function to get the schema for a given workflow
+        :param wf: workflow name
+        :param drop_null: if True, drop key with null values from the schema
+        :param make_copy: if True, return a copy of the schema
+        :param from_cache: if True, get schema from cache
+        :return: a dict
+        """
+        result = getResponse(
+            url=self.reqmgrUrl,
+            endpoint=self.reqmgrEndpoint["couchdb" if from_cache else "req"] + wf,
+        )
+        if make_copy:
+            data = copy.deepcopy(result["result"][0][wf])
+        else:
+            data = result["result"][0][wf]
+            
+        if drop_null:
+            return {k: v for k, v in data.items() if v not in [None, "None"]}
+        return data
+
+    def getWorkflowSchemaByCampaign(
+        self, campaign: str, details: bool = False
+    ) -> Union[List[dict], List[str]]:
         """
         The function to get the list of workflows for a given campaign
         :param campaign: campaign name
         :param details: if True, it returns details for each workflow, o/w, just workflow names
         :return: list of dicts if details True, list of strings o/w
         """
+        try:
+            return self._getWorkflowSchemaByParam({"campaign": campaign}, details)
 
+        except Exception as error:
+            self.logger.error(
+                f"Failed to get workflows from reqmgr for campaign {campaign}"
+            )
+            self.logger.error(str(error))
+
+    def getWorkflowSchemaByOutput(
+        self, dataset: str, details: bool = False
+    ) -> Union[List[dict], List[str]]:
+        """
+        The function to get the list of workflows for a given output
+        :param dataset: dateset name
+        :param details: if True, it returns details for each workflow, o/w just workflows names
+        :return: a list of dicts if details is True, list of strings o/w
+        """
         try:
-            result = getResponse(url=self.reqmgrUrl,
-                                 endpoint='/reqmgr2/data/request/',
-                                 param={"campaign": campaign, "detail": str(details)})
+            return self._getWorkflowSchemaByParam({"outputdataset": dataset}, details)
 
+        except Exception as error:
+            self.logger.error(
+                f"Failed to get workflows from reqmgr for dataset {dataset}"
+            )
+            self.logger.error(str(error))
+
+    def getWorkflowSchemaByPrepId(
```

Comment:
Change name to `getWorkflowsByPrepId`. `Schema` is confusing in case `details` is False.